node/actors/control_api.go

```diff
@@ -181,62 +180,198 @@ func (api *ControlAPI) subscribe() error {
 }
 
 func (api *ControlAPI) handleAuction(m *nats.Msg) {
-	// TODO
+	req := new(nodecontrol.AuctionRequestJson)
+	err := json.Unmarshal(m.Data, req)
+	if err != nil {
+		api.logger.Error("Failed to unmarshal auction request", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to unmarshal auction request: %s", err))
+		return
+	}
+
+	convertedAgentType := make([]string, len(req.AgentType))
+	for _, at := range req.AgentType {
+		convertedAgentType = append(convertedAgentType, string(at))
+	}
+
+	auctResp, err := api.auctionResponse(string(req.Os), string(req.Arch), convertedAgentType, req.Tags.Tags)
+	if err != nil {
+		api.logger.Error("Failed to generate auction response", slog.Any("error", err))
+		respondEnvelope(m, AuctionResponseType, 500, nil, fmt.Sprintf("failed to generate auction response: %s", err))
+		return
+	}
+
+	if auctResp == nil {
+		respondEnvelope(m, AuctionResponseType, 404, nil, "node does not satisfy auction request")
```

Comment:
if we don't pass the auction, we need to not respond to the auction at all. In a scatter/gather like this, we can't respond negatively.